games-roguelike/nethack/nethack-3.6.1.ebuild

```diff
@@ -0,0 +1,135 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit desktop toolchain-funcs flag-o-matic user
+
+MY_PV=${PV//.}
+DESCRIPTION="The ultimate old-school single player dungeon exploration game"
+HOMEPAGE="https://www.nethack.org/"
+SRC_URI="https://nethack.org/download/${PV}/nethack-${MY_PV}-src.tgz"
+
+LICENSE="nethack"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~hppa ~x86 ~x86-fbsd"
+IUSE="X"
+
+RDEPEND="sys-libs/ncurses:0=
+	X? (
+		x11-libs/libXaw
+		x11-libs/libXpm
+		x11-libs/libXt
+	)"
+DEPEND="${RDEPEND}
+	X? ( x11-base/xorg-proto )
+	"
+BDEPEND="virtual/pkgconfig
+	X? (
+		x11-apps/bdftopcf
+		x11-apps/mkfontdir
+	)"
+
+BINDIR="/usr/bin"
+STATEDIR="/var/games/${PN}"
+
+NETHACK_GROUP="gamestat"
+
+pkg_setup() {
+	HACKDIR="/usr/$(get_libdir)/${PN}"
+
+	enewgroup gamestat 36
+}
+
+src_prepare() {
+	eapply "${FILESDIR}/${P}-recover.patch"
+	eapply_user
+
+	cp "${FILESDIR}/${PN}-3.6.0-hint-$(usex X x11 tty)" hint || die "Failed to copy hint file"
+	sys/unix/setup.sh hint || die "Failed to run setup.sh"
+}
+
+src_compile() {
+	append-cflags -I../include -DDLB -DSECURE -DLINUX -DTIMED_DELAY -DVISION_TABLES
+	append-cflags -DDUMPLOG -DSCORE_ON_BOTL -DSTATUS_HILITES
+	append-cflags '-DCOMPRESS=\"/bin/gzip\"' '-DCOMPRESS_EXTENSION=\".gz\"'
+	append-cflags "-DHACKDIR=\\\"${HACKDIR}\\\"" "-DVAR_PLAYGROUND=\\\"${STATEDIR}\\\""
+	append-cflags "-DDEF_PAGER=\\\"${PAGER}\\\""
+	append-cflags -DSYSCF "-DSYSCF_FILE=\\\"/etc/nethack.sysconf\\\""
+
+	use X && append-cflags -DX11_GRAPHICS -DUSE_XPM
+
+	makeopts=(
+		CC="$(tc-getCC)" CFLAGS="${CFLAGS}" LFLAGS="${LDFLAGS}"
+		WINTTYLIB="$($(tc-getPKG_CONFIG) --libs ncurses)"
+		HACKDIR="${HACKDIR}" INSTDIR="${D}/${HACKDIR}"
+		SHELLDIR="${D}/${BINDIR}" VARDIR="${D}/${STATEDIR}"
+		)
+
+	emake "${makeopts[@]}" nethack recover Guidebook spec_levs
+
+	# Upstream still has some parallel compilation bugs
+	emake -j1 "${makeopts[@]}" all
+}
+
+src_install() {
+	emake "${makeopts[@]}" install
+
+	exeinto "${BINDIR}"
+	newexe util/recover recover-nethack
+	rm "${D}/${HACKDIR}/recover" || die "Failed to remove HACKDIR/recover"
```

Comment:
Why not just rename the installed file? That seems cleaner than installing it again just to remove the previous copy.